pkg/job/job.go

```diff
@@ -85,3 +85,10 @@ func (j *Job) IsCancelled() bool {
 		return false
 	}
 }
+
+// InfoFetcher defines how to fetch job information
+type InfoFetcher interface {
```

Comment:
`JobFetcher` was my initial iteration. Eventually I used `job.InfoFetcher` because this interface defines methods to get job instancess and other job information (like the job IDs), and I had in mind of adding more job info-related methods. If you think this is confusing I can change its name, but I'm not convinced by `JobFetcher` because it's more narrow in scope than how I plan to use this interface